File: BackendPersonalFinanceTracker/FinanceTracker.Data/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;

namespace FinanceTracker.Data.Models
{
    public class Transaction
    {
        [Key]
        public int TransactionId { get; set; }
        
        [Required]
        public string Title { get; set; }

        [Required]
        public DateTime TransactionDate { get; set; }

        public int? TransactionDescriptionId { get; set; }

        [Required]
        public decimal Price { get; set; }

        [Required]
        public bool IsIncome { get; set; }

        public virtual TransactionDescription? TransactionDescription { get; set; }
    }
}

File: WebPersonalFinanceTracker/src/app/interfaces/transaction.ts
export class Transaction {
  transactionId!: number;
  title!: string;
  transactionDate!: DateTime;
  price!: number;
  isIncome!: boolean;
  transactionDescriptionName!: string | null;

  get fullDateString(): string {
    return `${this.transactionDate.day}/${this.transactionDate.month}/${this.transactionDate.year} ${this.transactionDate.hour}:${this.transactionDate.minute}`;
  }
}

export class DateTime {
  year!: number;
  month!: number;
  day!: number;
  hour!: number;
  minute!: number;
}
